src/main/java/com/venafi/vcert/sdk/policy/converter/tpp/TPPPolicySpecificationValidator.java

```diff
@@ -0,0 +1,183 @@
+
+                if(policyKeyPair != null) {
+                    String[] policyKeyTypes = policyKeyPair.keyTypes();
+                    if (policyKeyTypes != null && policyKeyTypes.length == 1 && !policyKeyTypes[0].equals("") && !policyKeyTypes[0].equals(defaultKeyType))
+                        throw new VCertException(String.format(DEFAULT_ATTRIBUTE_DOESNT_MATCH_EXCEPTION_MESSAGE, PolicySpecificationConst.ATT_DEFAULTS_KEYPAIR_KEY_TYPE, PolicySpecificationConst.ATT_POLICY_KEYPAIR_KEY_TYPES));
+                }
+            }
+
+            Integer defaultRsaKeySize = defaultsKeyPair.rsaKeySize();
+            if( defaultRsaKeySize != null ) {
+                if( !TPPKeyPairEnums.containsRsaKeySize( defaultRsaKeySize ))
+                    throw new VCertException(String.format(DEFAULT_ATTRIBUTE_DOESNT_MATCH_WITH_ACCEPTED_VALUES_EXCEPTION_MESSAGE, PolicySpecificationConst.ATT_DEFAULTS_KEYPAIR_RSA_KEY_SIZE));
+
+                if(policyKeyPair != null) {
+                    Integer[] policyRsaKeySizes = policyKeyPair.rsaKeySizes();
+                    if (policyRsaKeySizes != null && policyRsaKeySizes.length == 1 && !policyRsaKeySizes[0].equals(defaultRsaKeySize))
+                        throw new VCertException(String.format(DEFAULT_ATTRIBUTE_DOESNT_MATCH_EXCEPTION_MESSAGE, PolicySpecificationConst.ATT_DEFAULTS_KEYPAIR_RSA_KEY_SIZE, PolicySpecificationConst.ATT_POLICY_KEYPAIR_RSA_KEY_SIZES));
+                }
+            }
+
+            String defaultEC = defaultsKeyPair.ellipticCurve();
+            if ( defaultEC != null && !defaultEC.equals("")){
+                if ( !TPPKeyPairEnums.containsEllipticCurve( defaultEC ) )
+                    throw new VCertException(String.format(DEFAULT_ATTRIBUTE_DOESNT_MATCH_WITH_ACCEPTED_VALUES_EXCEPTION_MESSAGE, PolicySpecificationConst.ATT_DEFAULTS_KEYPAIR_ELLIPTIC_CURVE));
+
+                if(policyKeyPair != null) {
+                    String[] policyEC = policyKeyPair.ellipticCurves();
+                    if (policyEC != null && policyEC.length == 1 && !policyEC[0].equals("") && !policyEC[0].equals(defaultEC))
+                        throw new VCertException(String.format(DEFAULT_ATTRIBUTE_DOESNT_MATCH_EXCEPTION_MESSAGE, PolicySpecificationConst.ATT_DEFAULTS_KEYPAIR_ELLIPTIC_CURVE, PolicySpecificationConst.ATT_POLICY_KEYPAIR_ELLIPTIC_CURVES));
+                }
+            }
+
+            if(policyKeyPair != null
+                    && policyKeyPair.serviceGenerated() != null && defaultsKeyPair.serviceGenerated() != null
+                    && !policyKeyPair.serviceGenerated().equals(defaultsKeyPair.serviceGenerated())) {
+                    throw new VCertException(String.format(DEFAULT_ATTRIBUTE_DOESNT_MATCH_EXCEPTION_MESSAGE, PolicySpecificationConst.ATT_DEFAULTS_KEYPAIR_SERVICE_GENERATED_TYPE, PolicySpecificationConst.ATT_POLICY_KEYPAIR_SERVICE_GENERATED_TYPE));
+            }
+        }
+    }
+
+}
```

Comment:
No new line at end of file

Reply:
In fact that doesn't affect the functionallity of the class